lib/molinillo/dependency_graph/log.rb

```diff
@@ -0,0 +1,136 @@
+require 'molinillo/dependency_graph/action'
+
+module Molinillo
+  class DependencyGraph
+    class Log
+      def self.action(name__, parameters = nil, &blk)
+        name__ = name__.to_sym
+        cls = Class.new(Action) do
+          define_singleton_method(:name) { name__ }
+          if parameters || (method = begin
+                                       DependencyGraph.instance_method(name__)
+                                     rescue
+                                       nil
+                                     end)
+            parameters ||= method.parameters.map(&:last)
+            module_eval <<-EOS
+              #{parameters.map { |param| "attr_reader(:#{param});" }.join}
+              define_method(:initialize) do |#{parameters.join(', ')}|
+                #{parameters.map { |p| "@#{p} = #{p};" }.join}
+                @args = [#{parameters.join(', ')}]
+              end
+            EOS
+          end
+          module_exec(&blk)
```

Comment:
If for whatever reason you really need to eval a string, then it's best for debuggability that you include the file and line where this was defined.